src/classes/CRLP_Operation_SVC.cls

```diff
@@ -247,10 +253,16 @@ public class CRLP_Operation_SVC {
 
                 // If the result field is a currency type field and multi-currency is enabled in the org
                 // then convert it to the target currency
-                if (resultVal != null && r.resultFieldDisplayType == DisplayType.CURRENCY &&
-                    UTIL_Currency.getInstance().isMultiCurrencyOrganization()) {
-                    String fromCurrCode = UserInfo.getDefaultCurrency(); // default just in case there's an issue getting the detail record currency
-                    resultVal = UTIL_CurrencyConversion.convertAmount((Decimal) resultVal, dateValue, fromCurrCode, rollup.currCode);
+                if (resultVal != null && r.resultFieldDisplayType == DisplayType.CURRENCY && isMultiCurrencyOrg) {
+                    String fromCurrency = UserInfo.getDefaultCurrency(); // default just in case there's an issue getting the detail record currency
+                    try {
+                        Object currencyIsoCode = (r.detailObject == oppObjectName ?
+                            oppSObject.get('CurrencyIsoCode') : detailSObject.get('CurrencyIsoCode'));
+                        if (currencyIsoCode != null) {
+                            fromCurrency = (String) currencyIsoCode;
+                        }
+                    } catch (Exception ex) { }
```

Comment:
@meena-sfdc @rponti-sforg This is the main change I made to address the issue with not converting a SingleResultRollup amount field. In the previous code it would *always* convert to the currency of the User as opposed to convert to the currency of the target record as it is supposed to do. I added lines 258